cmd/node/node.go

```diff
@@ -348,6 +352,25 @@ func (app SpacemeshApp) stopServices() {
 
 }
 
+func getEdIdentity() *signing.EdSigner {
+	// read key from file if exist
+	log.Info("Reading identity from file")
+	dataDir, err := filesystem.GetSpacemeshDataDirectoryPath()
+	if err != nil {
+		log.Error("Could not get data path err=%v", err)
+		return signing.NewEdSigner()
+	}
+
+	buff, e1 := ioutil.ReadFile( dataDir + "/identity.ed")
+	edSgn, e2 := signing.NewEdSignerFromBuffer(buff)
```

Comment:
this is not polite to use `buff` before you checked for errors (I know that `NewEdSignerFromBuffer` will handle an empty buffer but still...). First check that `e1` isn't nil before moving forward.

Reply:
Fixed